### PR TITLE
Review

### PR DESCRIPTION
**Pull request by andrasq**
pull request for review of 1.0.14-p1

### CODE REVIEW

File: lib/filewriter.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    this.write = function(str, cb) {
+        var self = this;
+        this._getLockedFd(function(err, fd) {
+            if (err) return cb(err);
+            // not much benefit to reusing a buffer for the writes
+            var buf = new Buffer(str);
+            fs.write(fd, buf, 0, buf.length, null, function(err, nb) {
+                flockSync(fd, "un");
+                if (err) return cb(err);
+                // hack: prevent buf from being gc`d too early
```
Comment (galniv) [inline lib/filewriter.js]: Why is this needed?

Comment (andrasq) [inline lib/filewriter.js]: _getLockedFd() is used to make writes atomic (allow concurrent writes via different file handles).
By default overlapping writes can overwrite and/or interleave the data.

Comment (galniv) [inline lib/filewriter.js]: How does that relate to `buf` being garbage collected too early? why do you need buf to stick around after `fs.write` completes?

File: lib/fputs.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    }
+
+    /**
+     * Write bulk data to the target.  Newline termination is not checked.
+     */
+    Fputs.prototype.write = function write( str, callback ) {
+        if (typeof str !== 'string') str = "" + str;
+
+        // merge writes into this.writesize sized data chunks
+        // it is assumed that all writes end with a newline (not checked)
```
Comment (galniv) [inline lib/fputs.js]: What would happen if they aren't?

File: lib/filewriter.js (hunk 1 above)
Comment (andrasq) [inline lib/filewriter.js]: oh, I see.  the "gc'd too early" hack/fix I saw elsewhere (used in a similar context, when passing a buffer to write),
and I assumed it was a work-around for an issue internal to node, and copied it just to be safe.

Comment (galniv) [inline lib/filewriter.js]: Um, ok. I don't think it makes much sense to blindly copy an admitted hack without knowing what it fixes. Maybe it's for an old version of node? maybe it's for an unrelated issue? maybe the person who wrote it was wrong?

File: lib/fputs.js (hunk 2 above)
Comment (andrasq) [inline lib/fputs.js]: they get concatenated and written in larger batches, as expected.  This is in fact how kds uses qfputs
(the newline separators between batches were added independently, for legibility)

The reason for the comment is that qfputs was designed for high-speed streaming of newline-terminated
strings, with each line written to the file atomically (ie, each record lands in the file with its newline terminator).
This invariant for the data consumer is maintained by the `fputs()` method, but `write()` ignores newlines
and thus could send partial records to the consumer of the data stream.

The note is a reminder for programmers who manually concatenate records that they must
ensure that records are batched in their entirety with their terminating newline for data consistency.

File: lib/filewriter.js (hunk 1 above)
Comment (andrasq) [inline lib/filewriter.js]: yeah, any of that is possible.  It seemed a problem easier to prevent than to diagnose
